proposals/record-structs.md

```diff
@@ -0,0 +1,301 @@
+# Record structs
+
+The syntax for a record struct is as follows:
+
+```antlr
+struct_record_declaration
+    : attributes? struct_modifier* 'partial'? 'record' 'struct' identifier type_parameter_list?
+      parameter_list? struct_interfaces? type_parameter_constraints_clause* record_struct_body
+    ;
+
+record_struct_body
+    : struct_body
+    | ';'
+    ;
+```
+
+Record struct types are value types, like other struct types. They implicitly inherit from the class `System.ValueType`.
+The modifiers and members of a record struct are subject to the same restrictions as those of structs
+(accessibility on type, modifiers on members, parameterless instance constructors,
+`base(...)` instance constructor initializers, definite assignment for `this` in constructor, destructors, ...).
+
+See https://github.com/dotnet/csharplang/blob/master/spec/structs.md
+
+But instance field declarations for a record struct are permitted to include variable initializers when there is a primary constructor.
```

Comment:
Personally, I would say no, they can't, you need to define a constructor, or use the primary constructor body (which we also need to add to the language).

Reply:
Actually, thinking about this a bit more, maybe I can be convinced either way.

Reply:
It's listed as an open question. I can go either way too.

---
In reply to: [553002951](https://github.com/dotnet/csharplang/pull/4307#discussion_r553002951) [](ancestors = 553002951)